doc/MiniLanguageTutorial.md

```diff
@@ -146,18 +183,20 @@ Two array types are equal if their field types are equal.
 
 Two struct types are equal if have the same number of fields, and each field has the same name and equal type, field-by-field.
 
-Two func types are equal if they are both impure or both not-impure, and they have the same number of argument types, and each argument type is equal, argument-by-argument, and the return types are equal (or neither has a return type).
+Two func types are equal if they use the same qualifiers, they have the same number of argument types, and each argument type is equal, argument-by-argument, and the return types are equal (or neither has a return type).
```

Comment:
Two func types are equal if they have the same purity, not qualifiers. A func being `public` can be equal to one that isn't.